src/codec/zstd/decoder.rs

```diff
@@ -14,6 +14,13 @@ impl ZstdDecoder {
             decoder: Unshared::new(Decoder::new().unwrap()),
         }
     }
+
+    pub(crate) fn new_with_dict(dictionary: &[u8]) -> Self {
+        let mut decoder = Decoder::with_dictionary(dictionary).unwrap();
```

Comment:
loading dict could fail, so unwrapping here isn't a good idea.

Reply:
No problem been changed, is it guaranteed that `new` and `new_with_params` can't panic?